### PR TITLE
fix(fslib): handle `.zip` in the middle of file name again

### PR DESCRIPTION
**Pull request by ratson**
**What's the problem this PR addresses?**

Path with `.zip` in the middle of the file name is not extracted correctly.

Closes https://github.com/yarnpkg/berry/issues/2975

**How did you fix it?**

Added 2 edge cases to the test.

...

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by merceyz** [comment]
For context that function was added in https://github.com/yarnpkg/berry/pull/2796 to speed up installs / replace this regex
```regex
.*?(?<!\/)\.zip(?=\/|$)
```
We need to verify that this fix doesn't regress the performance to the point where we could just use the regex

**Comment by ratson** [comment]
@merceyz 
code block 1 is new code, 
code block 2 is regex version, 

Before:
https://jsben.ch/CTkgN
Screenshot 2021-06-22 at 22-55-08 JSBEN CH Benchmarking for JavaScript -

After:
https://jsben.ch/b07Zt
Screenshot 2021-06-22 at 22-54-03 JSBEN CH Benchmarking for JavaScript -

**Review by merceyz — CHANGES_REQUESTED** [review]
Should be updated to handle an "unlimited" amount of `.zip` occurrences in a filename

**Comment by ratson** [comment]
@merceyz Which test case is in your mind? The change should have handled multiple `.zip` occurrences.

File: packages/yarnpkg-fslib/tests/ZipOpenFS.test.ts [hunk 1]
```diff
@@ -30,6 +30,8 @@ describe(`getArchivePart`, () => {
     [`./a/b/c/.zip`, null],
     [`./a/b/c/foo.zipp`, null],
     [`./a/b/c/foo.zip/bar/baz/qux.zip`, `./a/b/c/foo.zip`],
+    [`./a/b/c/foo.zip-bar.zip`, `./a/b/c/foo.zip-bar.zip`],
+    [`./a/b/c/foo.zip-bar.zip/bar/baz/qux.zip`, `./a/b/c/foo.zip-bar.zip`],
```
Comment (merceyz) [inline packages/yarnpkg-fslib/tests/ZipOpenFS.test.ts:34]:
```suggestion
    [`./a/b/c/foo.zip-bar.zip/bar/baz/qux.zip`, `./a/b/c/foo.zip-bar.zip`],
    [`./a/b/c/foo.zip-bar/foo.zip-bar/foo.zip-bar.zip/d`, `./a/b/c/foo.zip-bar/foo.zip-bar/foo.zip-bar.zip`],
```

**Review by merceyz — COMMENTED** [review]
It's an edge case but here is a failing test case

**Comment by ratson** [comment]
https://jsben.ch/oBqKF
Screenshot 2021-06-24 at 21-41-00 JSBEN CH Benchmarking for JavaScript -
